### PR TITLE
Send contacts to every device

### PR DESCRIPTION
**Pull request by e0ff**
This change is [Reviewable](https://reviewable.io/reviews/grayhatter/toxcore/13)

### CODE REVIEW

File: toxcore/MDevice.c [hunk 1]
```diff
@@ -550,11 +550,11 @@ static int sync_friend_recived(Tox *tox, uint32_t dev_num, uint8_t *real_pk, boo
...
-/** TODO DOCUMENT THIS FXN
- *
- *
- *
- *
+/**
+ * Creates the packet for syncing friends between devices and then sends the
+ * packet to each device.
+ * returns 0 on successs
+ * returns -1, -2, or -3 depending on the error
```
Comment (GrayHatter) [inline toxcore/MDevice.c:566]: for API documentation you need to enumerate each on it's own line

File: toxcore/MDevice.c [hunk 2]
```diff
@@ -570,31 +570,37 @@ static int actually_send_friend_list(Tox *tox, uint32_t dev_num)
...
 
-        memset(packet, 0, sizeof(packet));
+            memset(packet, 0, sizeof(packet));
 
-        packet[0] = PACKET_ID_MDEV_SYNC;
-        packet[1] = MDEV_SYNC_CONTACT_APPEND;
+            packet[0] = PACKET_ID_MDEV_SYNC;
 
-        memcpy(packet + 2, tox->m->friendlist[i].dev_list[0].real_pk, crypto_box_PUBLICKEYBYTES);
+            if(j == 0)
```
Comment (GrayHatter) [inline toxcore/MDevice.c:607]: Code needs to match 1TBS https://en.wikipedia.org/wiki/Indent_style#Variant:_1TBS

File: toxcore/MDevice.c [hunk 3]
```diff
@@ -570,31 +570,37 @@ static int actually_send_friend_list(Tox *tox, uint32_t dev_num)
...
-        }
+            memcpy(packet + 2, tox->m->friendlist[i].dev_list[j].real_pk, crypto_box_PUBLICKEYBYTES);
 
-        sync_friend_recived(tox, dev_num, tox->m->friendlist[i].dev_list[0].real_pk, 0);
+            if (!send_mdev_packet(tox, dev_num, packet, sizeof(packet))) {
+                printf("ERROR sending MDEV_SYNC_CONTACT_APPEND packet \n");
+                return -2;
+            }
+
+            sync_friend_recived(tox, dev_num, tox->m->friendlist[i].dev_list[j].real_pk, 0);
```
Comment (GrayHatter) [inline toxcore/MDevice.c:623]: replace the ending 0 with `j`

**Comment by e0ff** [comment]
I'll open a new pull request I don't know what happened.

**Pull request closed without merging**